src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java

```diff
@@ -131,7 +134,8 @@ public BucketMetadata updateBucketOwner(
             // Check Grants
             if (!restApiAccessService.isBucketAccessibleByUser(sessionIdRequired, sessionId, bucketName)) {
                 AuthenticatedUser user = restApiAccessService.getUserFromSessionId(sessionId);
-                if (!bucketGrantService.isTheUserGrantSufficientForTheCurrentTask(user, bucketName, admin.toString())) {
+                if (!grantRightsService.getResultingAccessTypeFromUserGrantsForBucketOperations(user, bucketName)
```

Comment:
by looking at getResultingAccessTypeFromUserGrantsForBucketOperations code, you are not considering if the admin is the owner of the bucket.

a bucket owner group should be considered as a group admin grant with priority 5

Another remark. we need to think about two group grants with the same priority.
1) should it be disallowed ? In that case, I think 1-10 scale is not enough
2) should it be allowed ? In that case, what do we do if a user is a member of 2 groups with the same grant priority ?